frontend/src/features/recommendation/RecommendationApi.tsx

```diff
@@ -48,30 +41,34 @@ export const getRecommendedVideos = (
     }
   }
 
-  params.append('limit', '20');
-  searchString = params.toString();
+  const getNumberValue = (key: string): number | undefined => {
+    const value = params.get(key);
+    return value ? Number(value) : undefined;
+  };
 
-  // TODO once backend is fixed, use automatically generated code
-  fetch(`${api_url}/video/?`.concat(searchString), {
-    // /?language=` + language + '&date=' + date if you wan to add parameters
-    method: 'GET',
-    mode: 'cors',
-    headers: {
-      'Content-Type': 'application/x-www-form-urlencoded',
-      Authorization: 'Basic ' + btoa(client_id + ':' + client_secret),
-    },
-  })
-    .then((response) => {
-      return response.json();
-    })
-    .then((data) => {
-      callback(data);
-    })
-    .catch((err) => {
-      console.log(err);
-      callback({
-        results: [],
-        count: 0,
-      });
+  try {
+    return await VideoService.videoList({
+      limit: 20,
+      offset: getNumberValue('offset'),
+      search: params.get('search') ?? undefined,
+      language: params.get('language') ?? undefined,
+      dateGte: params.get('date_gte') ?? undefined,
+      largelyRecommended: getNumberValue('largely_recommended'),
+      reliability: getNumberValue('reliability'),
+      importance: getNumberValue('importance'),
+      engaging: getNumberValue('engaging'),
+      pedagogy: getNumberValue('pedagogy'),
+      laymanFriendly: getNumberValue('layman_friendly'),
+      diversityInclusion: getNumberValue('diversity_inclusion'),
+      backfireRisk: getNumberValue('backfire_risk'),
+      betterHabits: getNumberValue('better_habits'),
+      entertainingRelaxing: getNumberValue('entertaining_relaxing'),
```

Comment:
I think this could be automatically generated from the `allCriteriaNames` constant. Unless Typescript complains about it?

Reply:
Done, using a new util function to convert params from `snake_case` to `camelCase`.